core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/DataAccessImpl.java

```diff
@@ -72,13 +96,157 @@
+    }
+
+    private enum TempStatement {
+        // Should I map these to something else..? More like a key to the actual statement
+        dateRangeExclusive(byDateRangeExclusiveBase, StatementType.READ),
+        dateRangeExclusiveWithLimit(dateRangeExclusiveWithLimitBase, StatementType.READ),
+        dataByDateRangeExclusiveASC(dataByDateRangeExclusiveASCBase, StatementType.READ),
+        dataByDateRangeExclusiveWithLimitASC(dataByDateRangeExclusiveWithLimitASCBase, StatementType.READ),
+        SCAN_WITH_TOKEN_RANGES(scanTableBase, StatementType.SCAN),
+        CHECK_EXISTENCE_OF_METRIC_IN_TABLE(findMetricInDataBase, StatementType.SCAN),
+        LIST_ALL_METRICS_FROM_TABLE(findAllMetricsInDataBases, StatementType.SCAN),
+        INSERT_DATA(data, StatementType.WRITE),
+        INSERT_DATA_WITH_TAGS(dataWithTags, StatementType.WRITE),
+        CREATE_TABLE(TEMP_TABLE_BASE_CREATE, StatementType.CREATE),
+//        CREATE_WAL(TEMP_TABLE_WAL_CREATE, StatementType.CREATE);
+        DELETE_DATA(DELETE_FROM_DATA_BASE, StatementType.DELETE);
+
+        private final String statement;
+        private StatementType type;
+
+        TempStatement(String st, StatementType t) {
+            statement = st;
+            type = t;
+        }
+
+        public String getStatement() {
+            return statement;
+        }
+
+        public StatementType getType() {
+            return type;
+        }
+    }
+
+    // Read statement prototypes
+
+    private static String byDateRangeExclusiveBase =
+            "SELECT time, %s, tags FROM %s " +
+                    "WHERE tenant_id = ? AND type = ? AND metric = ? AND dpart = ? AND time >= ? AND time < ?";
+
+    private static String dateRangeExclusiveWithLimitBase =
+            "SELECT time, %s, tags FROM %s " +
+                    " WHERE tenant_id = ? AND type = ? AND metric = ? AND dpart = ? AND time >= ? AND time < ?" +
+                    " LIMIT ?";
+
+    private static String dataByDateRangeExclusiveASCBase =
+            "SELECT time, %s, tags FROM %s " +
+                    "WHERE tenant_id = ? AND type = ? AND metric = ? AND dpart = ? AND time >= ?" +
+                    " AND time < ? ORDER BY time ASC";
+
+    private static String dataByDateRangeExclusiveWithLimitASCBase =
+            "SELECT time, %s, tags FROM %s" +
+                    " WHERE tenant_id = ? AND type = ? AND metric = ? AND dpart = ? AND time >= ?" +
+                    " AND time < ? ORDER BY time ASC" +
+                    " LIMIT ?";
+
+    private static String scanTableBase =
+            "SELECT tenant_id, type, metric, time, n_value, availability, l_value, tags, token(tenant_id, type, " +
+                    "metric, dpart) FROM %s " +
+                    "WHERE token(tenant_id, type, metric, dpart) > ? AND token(tenant_id, type, metric, dpart) <=" +
+                    " ?";
+
+    // Create statement prototype
+
+    private static String TEMP_TABLE_BASE_CREATE = "CREATE TABLE %s ( " +
```

Comment:
> It made no difference to store bigint or smallint

Are you talking about on size on disk? It should make a difference for heap memory usage when reading rows.